src/Core/Domain/Product/Supplier/ProductSupplierUpdate.php

```diff
@@ -29,18 +29,17 @@
 namespace PrestaShop\PrestaShop\Core\Domain\Product\Supplier;
 
 use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
-use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ValueObject\ProductSupplierId;
-use PrestaShop\PrestaShop\Core\Domain\Supplier\ValueObject\SupplierId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Supplier\ValueObject\SupplierAssociationInterface;
 
 /**
  * Transfers data of product supplier
  */
-class ProductSupplier
+class ProductSupplierUpdate
 {
     /**
-     * @var SupplierId
+     * @var SupplierAssociationInterface
      */
-    private $supplierId;
+    private $association;
```

Comment:
can we be more verbose and use 
```suggestion
    private $supplierAssociation;
```

Reply:
I made it match the getter `getAssociation` and I named it a bit shorter on purpose because this is in the `ProductSupplier` namespace. In the repository, I also have a `ProductSupplierRepository::getByAssociation` which uses the same short naming

I think the namespace + the strict type everywhere is explicit enough 🤷‍♂️

You think using just `association` is confusing?